socotratools/socotratools/client.py

```diff
@@ -352,3 +352,62 @@ def get_report(self, locator):
     def get_locator(self, display_id):
         data = {'displayId': display_id}
         return self.__get("/policy/locator", params=data)
+
+    def get_grace_lapse_reinstatements(self, locator):
+        return self.__get(
+            "/policy/{0}/graceLapseReinstatements".format(locator))
+
+    def is_lapsed(self, mods):
+            i = 0
+            lapse_counter = 0
+            reinstate_counter = 0
+
+            for mod in mods:
+                i = i + 1
+                if mod['name'] == 'modification.policy.lapse':
+                    lapse_counter = i
+                elif mod['name'] == 'modification.policy.reinstate':
+                    reinstate_counter = i
+
+            if lapse_counter > reinstate_counter:
+                return True
+            else:
+                return False
+
+    def is_graced(self, glrs):
+
+        for glr in glrs:
+            graceResponse = glr['gracePeriod']
+            if graceResponse.get('settledTimestamp'):
+                continue
+            else:
+                return True
+        return False
+
+    def is_finalized(self, mods):
```

Comment:
This definitely works, but instead of going through each modification, why not just find the modification for creation and check that one `modification.policy.create`?